### PR TITLE
#315 - refactor + change selection algorithm for simple objects

### PR DESCRIPTION
**Pull request by ensemenova**

### CODE REVIEW

File: packages/ketcher-react/src/script/render/restruct/index.js [hunk 1]
```diff
@@ -616,8 +616,21 @@ ReStruct.prototype.setSelection = function (selection) {
...
+/**
+ * SelectionPlate could be an item then value would be in it
+ * or it could be a set of items then removed value need to be check on at least one of items in set
+ * @param item
+ * @returns {boolean}
+ */
+function isSelectionSvgObjectExists(item) {
+  return (
+    (item.selectionPlate !== null && !item.selectionPlate.removed) ||
+    (item.selectionPlate instanceof Array && !item.selectionPlate[0].removed)
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/render/restruct/index.js:628]: better change
`item.selectionPlate instanceof Array` to `Array.isArray(item.selectionPlate)`

File: packages/ketcher-react/src/script/render/restruct/resimpleobject.js [hunk 2]
```diff
@@ -366,9 +343,23 @@ ReSimpleObject.prototype.makeSelectionPlate = function (
     return scale.obj2scaled(p, restruct.render.options) || new Vec2()
   })
 
-  const path = generatePath(this.item.mode, paper, pos)
-
-  return path.attr(styles.highlightStyleSimpleObject)
+  const refPoints = this.getReferencePoints()
+  const s = restruct.render.options.scale
+  var st = restruct.render.paper.set()
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/render/restruct/resimpleobject.js:348]: please add more meaningfull names insted of `s` and `st`

Comment (ensemenova) [reply packages/ketcher-react/src/script/render/restruct/resimpleobject.js]: fixed

File: packages/ketcher-react/src/script/render/restruct/index.js (hunk 1 above)
Comment (ensemenova) [reply packages/ketcher-react/src/script/render/restruct/index.js]: fixed

File: packages/ketcher-react/src/script/render/restruct/resimpleobject.js [hunk 3]
```diff
@@ -210,17 +210,17 @@ ReSimpleObject.prototype.highlightPath = function (render) {
         render.paper.ellipse(
           tfx(point[0].x + rx),
           tfx(point[0].y + ry),
-          tfx(Math.abs(rx) + s / 8),
-          tfx(Math.abs(ry) + s / 8)
+          tfx(Math.abs(rx) + scale / 8),
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/render/restruct/resimpleobject.js:213]: Are you shure about `scale` and not `scaleFactor`? I guess here is some mistake

Comment (ensemenova) [reply packages/ketcher-react/src/script/render/restruct/resimpleobject.js]: thank you, fixed

**Review by TomasLight — APPROVED** [review]

**Pull request merged**